### DISCUSSION TITLE
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xdf in position 16: invalid continuation byte

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [coin-or/Ipopt#625] [General]
Hi,

Please refer to this: https://github.com/mdolab/pyoptsparse/discussions/323#discussion-4728865

Please advise.

**Comment by svigerske** [comment]
It seems that mpi4py has some problem to read a file. I don't see how this is related to Ipopt at all.

**Reply by uttamcadambi07** [reply]
I do not get the error at every iteration. If I have a small problem size (27K design variables for example), I get this error at some Nth iteration. But if my problem size is larger (say 45K design variables), I end up getting the error on the first iteration. I do not see how mpi4py managed to handle a problem size of 27K upto some point and then decided to have an issue reading a file all of a sudden when the problem size almost doubled. This is why I was inclined to believe it maybe had something to do with IPOPT.

**Reply by svigerske** [reply]
The error comes from Python code that doesn't belong to Ipopt. I don't understand what you would expect the Ipopt devs to do about this.

I would suggest you send something that reproduces the issue to the pyOptSparse developers. Then they have something they can actually look into and maybe figure out where something is going wrong. If that is in Ipopt, then they will hopefully be able to produce an error report that the Ipopt devs can work with.

If the pyOptSparse developers cannot or will not look into this, then you'll have to dig into this by yourself.

**Reply by uttamcadambi07** [reply]
Right firstly apologies @svigerske for not sufficiently investigating the problem before posting it here. Indeed you were right - mpi4py does have an issue. It is to do with broadcasting. When I searched the mpi4py forums, I found two links that closely resembled my issue:
1. https://bitbucket.org/mpi4py/mpi4py/issues/39/mpi-msgpicklepxi-eoferror
2.  https://bitbucket.org/mpi4py/mpi4py/issues/39/mpi-msgpicklepxi-eoferror

Some key takebacks:
1. set export I_MPI_OFA_TRANSLATION_CACHE=off was one suggestion provided in these forums. 
2. Running the cases on a single node solved the problem (since there are no inter-node communications which is probably why I did not face the error on my work station). Currently this fix has worked for me. But I will try the other fixes to see if they are helpful and if not, I will compile and send the report to the pyOptSparse developers. 

Just for reference, this is the code snippet from the file pyOpt_optimizer.py lines 673-693
```
    def _waitLoop(self):
        """Non-root processors go into this waiting loop while the
        root proc does all the work in the optimization algorithm"""

        mode = None
        info = None
        while True:
            # * Note*: No checks for MPI here since this code is
            # * only run in parallel, which assumes mpi4py is working

            # Receive mode and quit if mode is -1:
            mode = self.optProb.comm.bcast(mode, root=0)
            if mode == -1:
                break

            # Otherwise receive info from shell function
            info = self.optProb.comm.bcast(info, root=0)

            # Call the generic internal function. We don't care
            # about return values on these procs
            self._masterFunc2(*info)
```
